--- functions.php	2010-08-02 19:11:50.000000000 +0200
+++ functions.php	2010-08-03 11:25:38.321776583 +0200
@@ -839,6 +839,16 @@
  */
 function multipage($count, $perpage, $page, $url, $breadcrumb=false)
 {
+	if(function_exists("google_seo_url_multipage"))
+	{
+		$newurl = google_seo_url_multipage($url);
+
+		if($newurl)
+		{
+			$url = $newurl;
+		}
+	}
+
 	global $theme, $templates, $lang, $mybb;
 
 	if($count <= $perpage)
@@ -4416,6 +4426,16 @@
  */
 function get_profile_link($uid=0)
 {
+	if(function_exists("google_seo_url_profile"))
+	{
+		$link = google_seo_url_profile($uid);
+
+		if($link)
+		{
+			return $link;
+		}
+	}
+
 	$link = str_replace("{uid}", $uid, PROFILE_URL);
 	return htmlspecialchars_uni($link);
 }
@@ -4428,6 +4448,16 @@
  */
 function get_announcement_link($aid=0)
 {
+	if(function_exists("google_seo_url_announcement"))
+	{
+		$link = google_seo_url_announcement($aid);
+
+		if($link)
+		{
+			return $link;
+		}
+	}
+
 	$link = str_replace("{aid}", $aid, ANNOUNCEMENT_URL);
 	return htmlspecialchars_uni($link);
 }
@@ -4481,6 +4511,16 @@
  */
 function get_forum_link($fid, $page=0)
 {
+	if(function_exists("google_seo_url_forum"))
+	{
+		$link = google_seo_url_forum($fid, $page);
+
+		if($link)
+		{
+			return $link;
+		}
+	}
+
 	if($page > 0)
 	{
 		$link = str_replace("{fid}", $fid, FORUM_URL_PAGED);
@@ -4504,6 +4544,16 @@
  */
 function get_thread_link($tid, $page=0, $action='')
 {
+	if(function_exists("google_seo_url_thread"))
+	{
+		$link = google_seo_url_thread($tid, $page, $action);
+
+		if($link)
+		{
+			return $link;
+		}
+	}
+
 	if($page > 1)
 	{
 		if($action)
@@ -4543,6 +4593,16 @@
  */
 function get_post_link($pid, $tid=0)
 {
+	if(function_exists("google_seo_url_post"))
+	{
+		$link = google_seo_url_post($pid, $tid);
+
+		if($link)
+		{
+			return $link;
+		}
+	}
+
 	if($tid > 0)
 	{
 		$link = str_replace("{tid}", $tid, THREAD_URL_POST);
@@ -4564,6 +4624,16 @@
  */
 function get_event_link($eid)
 {
+	if(function_exists("google_seo_url_event"))
+	{
+		$link = google_seo_url_event($eid);
+
+		if($link)
+		{
+			return $link;
+		}
+	}
+
 	$link = str_replace("{eid}", $eid, EVENT_URL);
 	return htmlspecialchars_uni($link);
 }
@@ -4579,6 +4649,16 @@
  */
 function get_calendar_link($calendar, $year=0, $month=0, $day=0)
 {
+	if(function_exists("google_seo_url_calendar"))
+	{
+		$link = google_seo_url_calendar($calendar, $year, $month, $day);
+
+		if($link)
+		{
+			return $link;
+		}
+	}
+
 	if($day > 0)
 	{
 		$link = str_replace("{month}", $month, CALENDAR_URL_DAY);
@@ -4617,6 +4697,16 @@
  */
 function get_calendar_week_link($calendar, $week)
 {
+	if(function_exists("google_seo_url_calendar_week"))
+	{
+		$link = google_seo_url_calendar_week($calendar, $week);
+
+		if($link)
+		{
+			return $link;
+		}
+	}
+
 	if($week < 0)
 	{
 		$week = str_replace('-', "n", $week);
